services-camel/src/test/java/edu/unc/lib/dl/services/camel/images/ImageEnhancementsRouterTest.java

```diff
@@ -123,40 +150,146 @@ public void testThumbDisallowedImageType() throws Exception {
 
+        Map<String, Object> headers = createEvent(fileID, eventTypes, "true");
         template.sendBodyAndHeaders("", headers);
 
         verify(addSmallThumbnailProcessor).process(any(Exchange.class));
         assertMockEndpointsSatisfied();
     }
 
     @Test
-    public void testThumbLargeRoute() throws Exception {
+    public void testThumbSmallRouteNoForceFileExists() throws Exception {
+        String derivativePath = idToPath(fileID, HASHED_PATH_DEPTH, HASHED_PATH_SIZE);
+        File existingFile = new File("target/" + derivativePath + "/" + fileID + ".png");
+        FileUtils.writeStringToFile(existingFile, "extracted text", "utf-8");
+
+        createContext(smallThumbRoute);
+
+        getMockEndpoint("mock:exec:/bin/sh").expectedMessageCount(0);
+
+        Map<String, Object> headers = createEvent(fileID, eventTypes, "false");
+        template.sendBodyAndHeaders("", headers);
+
+        verify(addSmallThumbnailProcessor, never()).process(any(Exchange.class));
+        assertMockEndpointsSatisfied();
+        existingFile.delete();
+    }
+
+    @Test
+    public void testThumbSmallRouteForceFileExists() throws Exception {
+        String derivativePath = idToPath(fileID, HASHED_PATH_DEPTH, HASHED_PATH_SIZE);
+        File existingFile = new File("target/" + derivativePath + "/" + fileID + ".png");
+        FileUtils.writeStringToFile(existingFile, "extracted text", "utf-8");
+
+        createContext(smallThumbRoute);
+
+        getMockEndpoint("mock:exec:/bin/sh").expectedMessageCount(1);
+
+        Map<String, Object> headers = createEvent(fileID, eventTypes, "true");
+        template.sendBodyAndHeaders("", headers);
+
+        verify(addSmallThumbnailProcessor).process(any(Exchange.class));
+        assertMockEndpointsSatisfied();
+        existingFile.delete();
+    }
+
+    @Test
+    public void testThumbLargeRouteNoForceNoFileExists() throws Exception {
         createContext(largeThumbRoute);
 
         getMockEndpoint("mock:exec:/bin/sh").expectedMessageCount(1);
 
-        Map<String, Object> headers = createEvent(fileID, eventTypes);
+        Map<String, Object> headers = createEvent(fileID, eventTypes, "false");
 
         template.sendBodyAndHeaders("", headers);
 
+        doThrow(Exception.class).when(addLargeThumbnailProcessor).process(any(Exchange.class));
+
         verify(addLargeThumbnailProcessor).process(any(Exchange.class));
         assertMockEndpointsSatisfied();
     }
 
     @Test
-    public void testAccessCopyRoute() throws Exception {
+    public void testThumbLargeRouteForceNoFileExists() throws Exception {
+        createContext(largeThumbRoute);
+
+        getMockEndpoint("mock:exec:/bin/sh").expectedMessageCount(1);
+
+        Map<String, Object> headers = createEvent(fileID, eventTypes, "true");
+
+        template.sendBodyAndHeaders("", headers);
+
+        doThrow(Exception.class).when(addLargeThumbnailProcessor).process(any(Exchange.class));
```

Comment:
Is this throw serving a purpose here? Doesn't seem like what you'd expect in a successful case